gcc/config/loongarch/loongarch.cc

```diff
@@ -3258,6 +3309,71 @@ loongarch_split_move_insn (rtx dest, rtx src, rtx insn)
 static HOST_WIDE_INT
 loongarch_constant_alignment (const_tree exp, HOST_WIDE_INT align);
 
+const char *
+loongarch_output_move_index (rtx x, machine_mode mode, bool ldr)
+{
+  static char buf[128] = {0};
+
+  int index = exact_log2 (GET_MODE_SIZE (mode));
+  if (!IN_RANGE (index, 0, 3))
+    return NULL;
+
+  struct loongarch_address_info info;
+  if ((loongarch_classify_address (&info, x, mode, false)
+       && !(info.type == ADDRESS_REG_REG))
+      || !loongarch_legitimate_address_p(mode, x, false))
+    return NULL;
+
+  const char *const insn[][4] =
+    {
+      {
+	"stx.b\t%%z1,%%0",
+	"stx.h\t%%z1,%%0",
+	"stx.w\t%%z1,%%0",
+	"stx.d\t%%z1,%%0",
+      },
+      {
+	"ldx.bu\t%%0,%%1",
+	"ldx.hu\t%%0,%%1",
+	"ldx.w\t%%0,%%1",
+	"ldx.d\t%%0,%%1",
+      }
+    };
+  snprintf (buf, sizeof (buf), insn[ldr][index]);
+  return buf;
```

Comment:
why not directly return the pointer to the const string? it seems there's no postprocessing of any kind so the `snprintf` call seems superfluous.